xprocess.py

```diff
@@ -270,14 +277,16 @@ def _xshow(self, tw):
     def _clean_up_resources(self):
         # file handles should always be closed
         # in order to avoid ResourceWarnings
-        for f in self._file_handles:
+        for f in self._resources["file_handles"]:
             f.close()
         # XProcessInfo objects and Popen objects have
         # a one to one relation, so we should wait on
         # procs exit status if termination signal has
         # been isued for that particular XProcessInfo
         # Object (subprocess requirement)
-        for (info, _), proc in zip(self._info_objects, self._popen_instances):
+        for (info, _), proc in zip(
```

Comment:
the new structure is not a enhancmenet

resource should be a list of a simple classes that contain all resources of a process, just moving attributes into dict keys is not a sensible change, - literally all this does it make the code to access it longer

Reply:
I recall taking some time to figure out all allocated resources when we had the resource leak problem back in #61. It would have been a lot quicker if all resources (`Popen `instances, `XProcessInfo` instances and file handles) were all neatly and explicitly grouped in a `resources` structure such as a dict. So in my opinion this is still an improvement.

That being said, I agree that keeping the resources in a separate class seems like a better way of going about this. If I understand correctly, you're proposing having something like `XProcessResources` where all resources of a given `XProcess` instance would be kept (?). If this is the case, we could even further separate concerns by moving the cleanup logic from `XProcess._clean_up_resources` to `XProcessResources.free`. @RonnyPfannschmidt

Reply:
Structurally I would recommend a context manager or a close method to match other common apis